model/vcs/metrics.yaml

```
@@ -0,0 +1,151 @@
+    type: metric
+    metric_name: vcs.change.lines_delta
+    brief: 'The number of lines added/removed in a change relative to the common ancestor with the target branch.'
+    note: 'This metric should be reported for each `vcs.line_change.type` value. For example if a change added 3 lines and removed 2 lines, instrumentation SHOULD report two measurements: 3 and 2 (both positive numbers).'
+    instrument: gauge
+    unit: "{line}"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.name
+        requirement_level: required
+      - ref: vcs.ref.type
+        requirement_level: required
+      - ref: vcs.line_change.type
+        requirement_level: required
+  - id: metric.vcs.ref.lines_delta
+    type: metric
+    metric_name: vcs.ref.lines_delta
+    brief: 'The number of lines added/removed in a ref (branch) relative to the previous day'
+    note: 'This metric should be reported for each `vcs.line_change.type` value. For example if a ref added 3 lines and removed 2 lines, instrumentation SHOULD report two measurements: 3 and 2 (both positive numbers).'
+    instrument: gauge
+    unit: "{line}"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.name
+        requirement_level: required
+      - ref: vcs.ref.type
+        requirement_level: required
+      - ref: vcs.line_change.type
+        requirement_level: required
+  - id: metric.vcs.change.revisions_delta
+    type: metric
+    metric_name: vcs.change.revisions_delta
+    brief: 'The number of revisions (commits) a change is ahead/behind the target branch of the change'
+    note: 'This metric should be reported for each `vcs.revision_delta.direction` value. For example if branch `a` is 3 commits behind and 2 commits ahead of the target branch, instrumentation SHOULD report two measurements: 3 and 2 (both positive numbers).'
+    instrument: gauge
+    unit: "{revision}"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.name
+        requirement_level: required
+      - ref: vcs.ref.type
+        requirement_level: required
+      - ref: vcs.revision_delta.direction
+        requirement_level: required
+  - id: metric.vcs.ref.revisions_delta
+    type: metric
+    metric_name: vcs.ref.revisions_delta
+    brief: 'The number of revisions (commits) a ref (branch) is ahead/behind the default branch (trunk)'
+    note: 'This metric should be reported for each `vcs.revision_delta.direction` value. For example if branch `a` is 3 commits behind and 2 commits ahead of `trunk`, instrumentation SHOULD report two measurements: 3 and 2 (both positive numbers).'
+    instrument: gauge
+    unit: "{revision}"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
```

I would add these two attributes to the `vcs.ref.revisions_delta` metric.
With this we can optionally add the `vcs.change.id` if we are reporting the metric for a change.
We explicitly state against which other branch we compare the revisions against (new attribute).
This could the default branch for other branches and the target branch for a change.

```suggestion
    attributes:
      - ref: vcs.change.id
        requirement_level:
          conditionally_required: if a change is associate with the ref.
      - ref: vcs.ref.compare_to
        requirement_level: required
      - ref: vcs.repository.url.full
```